improver/utilities/spatial.py

```diff
@@ -214,44 +233,93 @@ def create_difference_cube(self, cube, coord_name, diff_along_axis):
         for coord in cube.derived_coords:
             dims = cube.coord_dims(coord)
             diff_cube.add_aux_coord(coord.copy(), dims)
-
-        # Add metadata to indicate that a difference has been calculated.
-        # TODO: update metadata for difference and add metadata for gradient
-        #       when proper conventions have been agreed upon.
-        if not self.is_gradient:
-            cell_method = CellMethod(
-                "difference", coords=[coord_name], intervals="1 grid length"
-            )
-            diff_cube.add_cell_method(cell_method)
-            diff_cube.attributes["form_of_difference"] = "forward_difference"
-        diff_cube.rename("difference_of_" + cube.name())
         return diff_cube
 
-    def calculate_difference(self, cube, coord_axis):
+    def calculate_difference(self, cube, coord_name):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=5583761)